src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java

```diff
@@ -48,6 +53,11 @@ public DARModalDetailsDTO(Document darDocument){
         setPrincipalInvestigator(darDocument.getString(DarConstants.INVESTIGATOR));
         setInstitutionName(this.institutionName = darDocument.getString(DarConstants.INSTITUTION));
         setProjectTitle(this.projectTitle = darDocument.getString(DarConstants.PROJECT_TITLE));
+        setDepartment(this.department = darDocument.getString(DarConstants.DEPARTMENT));
+        setCity(this.city = darDocument.getString(DarConstants.CITY));
+        setCountry(this.country = darDocument.getString(DarConstants.COUNTRY));
+        setNihUserName(this.nihUserName = darDocument.getString(DarConstants.NIH_USERNAME));
+        setHavenihUserName(this.havenihUserName = getNihUserName() != null && !getNihUserName().isEmpty());
```

Comment:
it could be simplified by: StringUtils.isNotEmpty(getNihUserName()). So getNihUserName will be called once

Reply:
Sounds good, please verify if this new builder approach is better. Note that one of the constructors of this DTO was used in DataAccessRequestResource class, so i had to move that logic into  a service in order to not mix implementation info with the resource.